SingularAdData.js

```diff
@@ -0,0 +1,101 @@
+export class SingularAdData {
+
+    constructor(adPlatform, currency, revenue) {
+        this._values = {
+            ad_platform: adPlatform,
+            ad_currency: currency,
+            ad_revenue: revenue,
+            is_admon_revenue: true,
+            is_revenue_event: true,
+            ad_mediation_platform: adPlatform
+        };
+    }
+
+    getAdDataValues() {
+        return this._values;
+    }
+
+    setNetworkName(networkName) {
+        this._values = {
+            ...this._values,
+            ad_mediation_platform: networkName
+        };
+    }
+
+    setAdType(adType) {
+        this._values = {
+            ...this._values,
+            ad_type: adType
+        };
+    }
+
+    setGroupType(adGroupType) {
+        this._values = {
+            ...this._values,
+            ad_group_type: adGroupType
+        };
+    }
+
+    setImpressionId(impressionId) {
+        this._values = {
+            ...this._values,
+            ad_impression_id: impressionId
+        };
+    }
+
+    setAdPlacementName(adPlacementName) {
+        this._values = {
+            ...this._values,
+            ad_placement_name: adPlacementName
+        };
+    }
+
+    setAdUnitId(adUnitId) {
+        this._values = {
+            ...this._values,
+            ad_unit_id: adUnitId
+        };
+    }
+
+    setAdGroupId(adGroupId) {
+        this._values = {
+            ...this._values,
+            ad_group_id: adGroupId
+        };
+    }
+
+    setAdGroupName(adGroupName) {
+        this._values = {
+            ...this._values,
+            ad_group_name: adGroupName
+        };
+    }
+
+    setAdGroupPriority(adGroupPriority) {
+        this._values = {
+            ...this._values,
+            ad_group_priority: adGroupPriority
+        };
+    }
+
+    setPrecision(precision) {
+        this._values = {
+            ...this._values,
+            ad_precision: precision
+        };
+    }
+
+    setPlacementId(placementId) {
+        this._values = {
+            ...this._values,
+            ad_placement_id: placementId
+        };
+    }
+
+    hasRequiredParams() {
+        if (!this._values.ad_currency || !this._values.ad_platform || !this._values.ad_revenue) {
```

Comment:
Let's have the required params in a list, and iterate over them when checking.
We also need to check they are not an empty string.
You can see in android how it's done

Reply:
no need to check explicitly for empty/null/undefined string .. just if(string) works.